src/Native/System.Native/pal_networkchange.cpp

```diff
@@ -52,7 +51,11 @@ extern "C" NetworkChangeKind SystemNative_ReadSingleEvent(int32_t sock)
     char buffer[4096];
     iovec iov = {buffer, sizeof(buffer)};
     sockaddr_nl sanl;
+#if MSGHDR_REQUIRES_PADDING
+    msghdr msg = {reinterpret_cast<void*>(&sanl), sizeof(sockaddr_nl), &iov, 1, 0, NULL, 0, 0, 0};
+#else
     msghdr msg = {reinterpret_cast<void*>(&sanl), sizeof(sockaddr_nl), &iov, 1, NULL, 0, 0};
+#endif
```

Comment:
I may have gotten the order of the fields wrong... double-check that I assigned things correctly.